src/TextTruncate/TextTruncate.tsx

```diff
@@ -47,20 +50,34 @@ export default class TextTruncate extends React.PureComponent {
   }
 
   render() {
-    const {className, text, showMoreLabel, showLessLabel, maxCharsShown} = this.props;
+    const {className, text, showMoreLabel, showLessLabel, maxCharsShown, richText} = this.props;
     const {truncated} = this.state;
 
     if (text.length < maxCharsShown) {
       return (<div className={classnames(cssClass.CONTAINER, className)}>
-        {text}
+        {richText ? <RichText text={text} /> : text}
       </div>);
     }
 
+    if (richText) {
+      return (
+        <div>
+          {truncated ?
+            <div className={classnames(cssClass.CONTAINER, className)}>
+              <RichText text={`${this.truncate(text)}…`} /> <Button type="linkPlain" onClick={this.toggleTruncation} value={showMoreLabel} />
+            </div> :
+            <div className={classnames(cssClass.CONTAINER, className)}>
+              <RichText text={text} /> <Button type="linkPlain" onClick={(e) => this.toggleTruncation(e)} value={showLessLabel} />
+            </div>
+          }
+        </div>
+      );
+    }
```

Comment:
Why does the lower snippet have a different signature for the `onClick` handler?

Pushing the conditional logic inside would reduce the amount of code here. This could be one snippet with `truncated ? A : B` inside of the `text` and `value` props.

Reply:
+1

Reply:
Good suggestion - I was able to make this radically simpler.